Spurious, (seemingly) very occasional NaN in

Last seen in https://github.com/inducer/meshmode/pull/213/checks?check_run_id=2770076958:

```
=================================== FAILURES ===================================
_ test_flatten_unflatten[<array context factory for <pyopencl.Device 'pthread-Intel(R) Xeon(R) CPU E5-2673 v4 @ 2.30GHz' on 'Portable Computing Language'>] _
[gw0] linux -- Python 3.9.4 /home/runner/work/meshmode/meshmode/.miniforge3/envs/testing/bin/python3
Traceback (most recent call last):
  File "/home/runner/work/meshmode/meshmode/test/test_array.py", line 93, in test_flatten_unflatten
    assert flat_norm(c - c_round_trip) < 1.0e-8
AssertionError: assert nan < 1e-08
 +  where nan = flat_norm((MyContainer(name='flatten', mass=DOFArray((cl.Array([[0.00000000e+000],\n       [            nan],\n       [4.65971693e-...  1.38196601e-01,\n               -1.11022302e-16]]),))                                          ],\n      dtype=object)) - MyContainer(name='flatten', mass=DOFArray((cl.Array([[0.00000000e+000],\n       [            nan],\n       [4.65971693e-...  1.38196601e-01,\n               -1.11022302e-16]]),))                                          ],\n      dtype=object))))
=============================== warnings summary ===============================
```

See also #210.

cc @alexfikl

--- a/test/test_array.py
+++ b/test/test_array.py
@@ -24,8 +24,8 @@ from dataclasses import dataclass
 import pytest
 import numpy as np
 
-import meshmode         # noqa: F401
-from arraycontext import (  # noqa: F401
+from arraycontext import _acf           # noqa: F401
+from arraycontext import (              # noqa: F401
         pytest_generate_tests_for_pyopencl_array_context
         as pytest_generate_tests)
 
@@ -80,7 +80,7 @@ def test_flatten_unflatten(actx_factory):
 
     x = thaw(discr.nodes(), actx)
     avg_mass = DOFArray(actx, tuple([
-        actx.empty((grp.nelements, 1), a.dtype) for grp in discr.groups
+        (np.pi + actx.zeros((grp.nelements, 1), a.dtype)) for grp in discr.groups
         ]))
 
     c = MyContainer(name="flatten",

